src/topology/algebra/module/locally_convex.lean

```diff
@@ -0,0 +1,138 @@
+/-
+Copyright (c) 2022 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+import analysis.convex.topology
+import analysis.seminorm
+/-!
+# Locally convex topological modules
+
+A `locally_convex_space` is a topological semimodule over an ordered semiring in which any point
+admits a neighborhood basis made of convex sets, or equivalently, in which convex neighborhoods of
+a point form a neighborhood basis at that point.
+
+In a module, this is equivalent to `0` satisfying such properties.
+
+## Main results
+
+- `locally_convex_space_iff_zero` : in a module, local convexity at zero gives
+  local convexity everywhere
+- `seminorm.locally_convex_space` : a topology generated by a family of seminorms is locally convex
+- `normed_space.locally_convex_space` : a normed space is locally convex
+
+## TODO
+
+- define a structure `locally_convex_filter_basis`, extending `module_filter_basis`, for filter
+  bases generating a locally convex topology
+- show that any locally convex topology is generated by a family of seminorms
+
+-/
+
+open topological_space filter
+
+open_locale topological_space
+
+section semimodule
+
+/-- A `locally_convex_space` is a topological semimodule over an ordered semiring in which convex
+neighborhoods of a point form a neighborhood basis at that point. -/
+class locally_convex_space (𝕂 E : Type*) [ordered_semiring 𝕂] [add_comm_monoid E] [module 𝕂 E]
+  [topological_space E] : Prop :=
+(convex_basis : ∀ x : E, (𝓝 x).has_basis (λ (s : set E), s ∈ 𝓝 x ∧ convex 𝕂 s) id)
```

Comment:
maybe rename it to `is_locally_convex`? This makes it easier to see that it is prop-valued.

Reply:
The thing is that we have for example `compact_space` and can't rename it to `is_compact` because that's the `set` version. Either way, it would be nice to coordinate with #11794.

Reply:
Yes I'm used to having both `is_X` and `X_space` for X a topological property, and that's why I named it that way. Of course here I don't think we'll ever want a set version, but for consistency I'd prefer keeping this name.

Yael, what you want to coordinate on specifically ?

Reply:
The name. We should call them `locally_convex_space`/`strict_convex_space` or `is_locally_convex_space`/`is_strict_convex_space` or ... but avoid disparity.

Reply:
The `is` version feels weird to me, but I let you choose

Reply:
I'm happy with `locally_convex_space`.

Reply:
yes, it is good. I looked around and it seems that the topology part of mathlib does not use `is_*` for spaces that much.